examples/tensorize_vllm_model.py

```diff
@@ -4,38 +4,106 @@
-    parser = AsyncEngineArgs.add_cli_args(parser)
+        description="Arguments for serializing and deserializing vLLM "
+                    "models. These models can be loaded using "
+                    "tensorizer directly to the GPU extremely quickly")
     parser = TensorizerArgs.add_cli_args(parser)
-    parser.add_argument("--serialize",
-                        default=False,
-                        action="store_true",
-                        help="If specified, serialize the model. "
-                        "Which will be saved to --tensorizer-uri.")
-    parser.add_argument("--deserialize",
-                        default=False,
-                        action="store_true",
-                        help="If specified, deserialize the model. "
-                        "Which will be loaded from --tensorizer-uri")
+    parser.add_argument(
+        "--model",
+        type=str,
+        required=True,
+        help="The model reference name to serialize or deserialize"
+    )
+    parser.add_argument(
+        "--dtype",
+        type=str,
+        required=False,
+        help="The dtype to use for the model"
+    )
+    parser.add_argument(
+        "--input-dir",
+        type=str,
+        required=True,
+        help="The directory to serialize or deserialize the model to. "
+             "For serialization, model tensors will be saved to this "
+             "directory under a directory given by the model ref set "
+             "in --model and a unique identifier. For instance, if "
+             "serializing EleutherAI/pythia-1.4b, in S3 bucket "
+             "BUCKET, tensors and encryption key if applicable will "
+             "be saved to "
+             "s3://BUCKET/EleutherAI/pythia-1.4b/HASH/model.tensors. "
+             "for this reason, --input-dir is recommended to specify "
+             "a bucket name if using object storage rather than a "
+             "local dir. If deserializing, model.tensors and "
+             "model.key will be looked for in --input-dir. In the "
+             "previous example, the --input-dir to use would be "
+             "s3://BUCKET/EleutherAI/pythia-1.4b/HASH"
+
+    )
+
+    subparsers = parser.add_subparsers(dest='command')
+
+    serialize_parser = subparsers.add_parser(
+        'serialize',
+        help="Serialize a model to `--input-dir`")
+
+    serialize_parser.add_argument(
+        "--keyfile",
+        type=str,
+        required=False,
+        help=("File to write 32 bytes of randomly-generated binary data used as"
+              " an encryption key"
+              )
+    )
+
+    deserialize_parser = subparsers.add_parser(
+        'deserialize',
+        help=(
+            "Deserialize a model from `--input-dir`"
+            " to verify it can be loaded and used."
+        )
+    )
+
+    deserialize_parser.add_argument(
+        "--keyfile",
+        type=str,
+        required=True,
+        help="Decryption keyfile to use to decrypt the model"
+    )
```

Comment:
`required` shouldn't be `True` if encryption is optional, and the description can be a little more precise.
```suggestion
    deserialize_parser.add_argument(
        "--keyfile",
        type=str,
        required=False,
        help=(
            "Path to a binary key to use to decrypt the model weights,"
            " if the model was serialized with encryption"
        )
    )
```